From 40bf99a41cb2cf387aceca7cec5f9c028ac125cb Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Wed, 8 Oct 2025 01:03:45 +0300
Subject: [PATCH 354/553] tst_Moc: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: I4a4afb120bc2e11a27cfcd7e15b1c9e5456f44da
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 7f6e19c69f4935f4c8dfe98860518bcf40854d31)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/tools/moc/tst_moc.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/auto/tools/moc/tst_moc.cpp b/tests/auto/tools/moc/tst_moc.cpp
index 217122100a0..95d6520a3a5 100644
--- a/tests/auto/tools/moc/tst_moc.cpp
+++ b/tests/auto/tools/moc/tst_moc.cpp
@@ -4282,7 +4282,7 @@ void tst_Moc::optionsFileError()
 }
 
 static void checkEnum(const QMetaEnum &enumerator, const QByteArray &name,
-                      const QList<QPair<QByteArray, quint64>> &keys,
+                      const QList<std::pair<QByteArray, quint64>> &keys,
                       const QMetaType enumType)
 {
     QCOMPARE(enumerator.name(), QByteArrayView{name});
@@ -4301,7 +4301,7 @@ static void checkEnum(const QMetaEnum &enumerator, const QByteArray &name,
 
 void tst_Moc::enumAndFlags64()
 {
-    const QList<QPair<QByteArray, quint64>> values = {
+    const QList<std::pair<QByteArray, quint64>> values = {
         { "Value0", 0 },
         { "ValueMixed", Q_UINT64_C(0x1122'3344'5566'7788) },
         { "ValueMinus1", quint64(-1) },
-- 
2.51.2

